src/ui/actions/users/unfollowUser.js

```
@@ -0,0 +1,12 @@
+import { test } from '@playwright/test';
+import { ProfilePage } from '../../pages/profile/ProfilePage';
+
+export async function unfollowUser(page, username) {
+  await test.step(`Unfollow user: ${username}`, async () => {
+    const profilePage = new ProfilePage(page);
+
+    await profilePage.open(username);
+    await profilePage.clickUnfollowButton();
+    await profilePage.assertFollowingState(false);
```

Behavioral fragility: the helper clicks `clickUnfollowButton()` unconditionally and then asserts the following state is false. If the profile was not followed before this action (i.e. the button currently shows "Follow"), clicking will actually follow the user and the assertion may fail or produce a false result. Make the helper resilient by one of the following approaches:

- Ensure precondition in caller tests: the caller must follow the user before calling this helper (call `clickFollowButton()` / assertFollowingState(true)).
- Make the helper check the button state and only click when it currently indicates "Unfollow" (or assert current state and fail early with a clear message).

You can rely on the existing `assertFollowingState` and `followButton` locator in ProfilePage to implement a safe check before clicking. See ProfilePage methods for reference (click/assert implementations) .